airflow/providers/cncf/kubernetes/backcompat/backwards_compat_converters.py

```diff
@@ -21,18 +21,16 @@
 from kubernetes.client import ApiClient, models as k8s
 
 from airflow.exceptions import AirflowException
-from airflow.providers.cncf.kubernetes.backcompat.pod import Port, Resources
-from airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info_env import PodRuntimeInfoEnv
 
 
-def _convert_kube_model_object(obj, old_class, new_class):
+def _convert_kube_model_object(obj, old_class_name, new_class):
```

Comment:
i'm confused why parameter `old_class_name` is here at all.  do you know?  it does not appear it has  any effect on the function's behavior

could we instead just remove this param?

Reply:
Its to print exception below only

Reply:
Of course I could remove mention of old class altogether so that user is not even considering it.

Reply:
it seems odd that the exception is, in part,`expected {old class name}`, when we do not appear to _check_ anywhere that it is `old_class_name`.... is that actually being checked?  if not why are we indicating that in the error?

Reply:
> Of course I could remove mention of old class altogether so that user is not even considering it

i'm not saying for certain that it should be removed, just taking a look and trying to understand.  i'll have a closer look at the code but don't consider my question in any way to be a blocker.

Reply:
We're converting it on the fly. So old class is not a blocker but rather a suggestion if user provided something completely unrelated.

Reply:
OK I think I get it now.  

We  used to use our own classes representing k8s objects.  Now we use the "official" ones.

For _our_ versions, we  have a conversion method `to_k8s_client_obj`

And we want a deprecation warning raised  when and only when we actually have to convert the object.

Ok so here's my suggestion.

I can think of two approaches.

1. add depreciation just before line  29 where `obj. to_k8s_client_obj()` is called.  In this case we know we are converting and can appropriately raise the warning.

or...

2. Add a deprecation warning to every `to_k8s_client_obj` so that whenever it is called, there's a warning.  

or 

3. add a base K8sObject class with a `to_k8s_client_obj` that emits the warning and ensure every backcompat airflow object  inherits from it and calls `super(). to_k8s_client_obj`

With either of these approaches, we don't need to import the backcompat modules.  We don't really need to know (or specify) the specific backcompat class expected in `_convert_kube_model_object` -- we can just explain we expected the k8s  object X or a backcompat  class appropriately convertible to X.

this would be my suggestion.

Reply:
I've applied alternative approach in separate - it will show warnings only if user imports the class:
https://github.com/apache/airflow/pull/20031
This specific modification we do here is to actually stop printing deprecation when its not present (right now it will print it always, even if we don't use old classes - just because of how project was structured)

Reply:
Your comment reminds me that by the time we get to this bit of code (e.g. if an airflow backcompat class is used) then  the  user  will _already_ have imported the backcompat module, and the warning presumably would have already been emitted. So no need, I suppose, to add another warning here as I have suggested in option 1.

That said I would still just get rid of the `old_class_name` param  since it isn't doing anything useful.  We can just say we expected class `new_class` or the equivalent airflow backcompat class..

Reply:
Fully reasonable. Okay, I removed the argument altogether and instead changed recommendation to only use new class. Whoever is using the old class will get necessary deprecation warning from "imports" and the logic will blindly convert old classes. However, if user provides something unrelated, it will recommend only new class to use (which is okay anyway).